### DISCUSSION TITLE
Let us `just deploy` a prebuilt project

### DISCUSSION THREAD

**Discussion by Fxlr8** [vercel/next.js#31157] [Ideas]
Hello. 
The git integration workflow and zero-config deployment features are great. They cover a bunch of use cases and save a lot of time. Unfortunately, not in my situation.

I use a self-managed GitLab instance, and I have a rather big CI/CD pipeline. It builds my Next.js app, runs unit and end-to-end tests, a performance score, uploads sourcemaps to Sentry, and only after that it begins the deployment step. I use Vercel CLI to deploy the preview app. However, every time I run the `vercel` command it uploads all my source code to your build server and builds the whole project again. I don't like it for two reasons:

1. I don't want to share the source code with third parties
2. I don't want to support and sync two independent build environments

I like Vercel's serverless infrastructure, your edge network, and the fact that I don't have to think about CDNs. But those two points make it hard for developers like me to become your customers.

I spent about a week tinkering and hacking around with the deployment process and I finally did it. I deployed a Next.js app that was built on my side to Vercel. I wrote a [Medium article](https://medium.com/@gturkin/deploy-a-prebuilt-next-js-app-to-vercel-51c5086655b9) on how I accomplished it.

Dear Vercel developers, please let us opt out of the git workflow and just deploy a prebuilt Next.js project to your network. So that we could use your service as a preview/deploy step of the pipeline. Not as a replacement of the whole CI/CD process.

**Comment by GodGardensGuns** [comment]
[The official approach](https://vercel.com/guides/deploying-locally-built-nextjs) is working for me with a Next.js + Turborepo project.
